tests/dialogs-controllers.tests.js

```diff
@@ -0,0 +1,431 @@
+        spyOn($scope, '$destroy');
+        spyOn($modalInstance, 'dismiss');
+
+      });
+    });
+
+    describe('Using Default Data', function(){
+      beforeEach(function(){
+        data = {};
+
+        $ctrlr('confirmDialogCtrl', {
+          $scope: $scope,
+          $modalInstance: $modalInstance,
+          $translate: $translate,
+          data: data
+        });
+      });
+
+      it('should dismiss modalInstance when no method executed', function(){
+        $scope.no();
+        expect($modalInstance.dismiss).toHaveBeenCalledWith('no');
+      });
+
+      it('should close modalInstance when yes method executed', function(){
+        $scope.yes();
+        expect($modalInstance.close).toHaveBeenCalledWith('yes');
+      });
+
+      it('should have default values for header', function(){
+        expect($translate.instant).toHaveBeenCalledWith('DIALOGS_CONFIRMATION');
+        expect($scope.header).toBe("Confirmation");
+      });
+
+      it('should have default values for msg', function(){
+        expect($translate.instant).toHaveBeenCalledWith('DIALOGS_CONFIRMATION_MSG');
+        expect($scope.msg).toBe("Confirmation required.");
+      });
+
+      it('should have default values for icon', function(){
+        expect($scope.icon).toBe("glyphicon glyphicon-check");
+      });
+
+    });
+
+    describe('Using Provided Data', function(){
+      beforeEach(function(){
+        data = {
+          header: "ThisConfirm",
+          msg: "Some Confirm Message",
+          fa: true
+        };
+
+        $ctrlr('confirmDialogCtrl', {
+          $scope: $scope,
+          $modalInstance: $modalInstance,
+          $translate: $translate,
+          data: data
+        });
+      });
+
+      it('should have ThisError for header', function(){
+        expect($scope.header).toBe("ThisConfirm");
+      });
+
+      it('should have Some Confirm Message for msg', function(){
+        expect($scope.msg).toBe("Some Confirm Message");
+      });
+
+      it('should have font-awesome values for icon', function(){
+        expect($scope.icon).toBe("fa fa-check");
+      });
+
+    });
+
+  });
+
+  describe('Controller waitDialogCtrl', function() {
+
+    var $scope;
+    var data;
+    var $ctrlr;
+    var $rootScope;
+
+    beforeEach(function(){
+      inject(function(_$rootScope_, $controller, $injector){
+
+        $ctrlr = $controller;
+        $scope = _$rootScope_.$new();
+        $rootScope = _$rootScope_;
+
+        spyOn($modalInstance, 'close');
+        spyOn($scope, '$destroy');
+        // spyOn($modalInstance, 'dismiss');
+
+      });
+    });
+
+    describe('Sending Events to $scope', function(){
+      beforeEach(function(){
+        $ctrlr('waitDialogCtrl', {
+          $scope: $scope,
+          $modalInstance: $modalInstance,
+          $translate: $translate,
+          data: {}
+        });
+      });
+
+      // it('should close when receiving event (dialogs.wait.complete) ', function(){
```

Comment:
I actually forgot i had commented this out, i was working on testing the broadcasts but i couldn't get them working, so i commented it out for now i'll just remove them.